### ISSUE TITLE
Multiple inheritance with NamedTuple is not supported, even when other class is Generic[T]

### ISSUE THREAD

**Issue by cpitclaudel** [microsoft/pyright#2203]
**Describe the bug**

Pyright complains about multiple inheritance with `NamedTuple` even when the other class is `Generic[T]`; shouldn't that be valid?  Otherwise, how does one typecheck generic tuples?

**To Reproduce**

```console
$ cat test.py
from typing import TypeVar, NamedTuple, Generic

X = TypeVar("X")
Y = TypeVar("Y")

class Pair(NamedTuple, Generic[X, Y]):
    x: X
    y: Y

$ pyright test.py
No configuration file found.
No pyproject.toml file found.
stubPath /tmp/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/tmp/test.py
  /tmp/test.py:6:7 - error: Multiple inheritance with NamedTuple is not supported (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos 
Completed in 0.647sec

$ mypy test.py
Success: no issues found in 1 source file

$ pyright --version
pyright 1.1.162
```

**Comment by erictraut** [comment]
This unfortunately isn't supported by the current `NamedTuple` implementation. If you run the above code in the Python interpreter, it results in the following error:

```
TypeError: Multiple inheritance with NamedTuple is not supported
```

**Comment by erictraut** [comment]
You may find this thread useful: https://stackoverflow.com/questions/50530959/generic-namedtuple-in-python-3-6

Can you use a dataclass instead? They support generics.

**Comment by cpitclaudel** [comment]
> Can you use a dataclass instead? They support generics.

Yes, I can use dataclasses.  Thanks for finding this thread, sorry for my poor search-engine-fu.

> This unfortunately isn't supported by the current NamedTuple implementation. 

Gah, sorry: it works in 3.7 and 3.8 (hence my report), but indeed not in 3.6, 3.9, and 3.10 (!)

**Comment by brandon-leapyear** [comment]
Python bug report: https://bugs.python.org/issue43923

**Comment by erictraut** [comment]
Yeah, that's an abuse of NamedTuple and TypeVar scoping. The TypeVar `T` is scoped to the outer `Foo`, but you're using it in the definition of the inner (dynamically-allocated) `Foo` NamedTuple.
